routes/profile.js

```diff
@@ -1,136 +1,118 @@
-  if (cv !== null && cv < 20) appendedRow.is_reliable = true;
-  if (comparison_cv !== null && comparison_cv < 20) appendedRow.comparison_is_reliable = true;
-
-  // set reliability to false if the value is top or bottom-coded
-  if (codingThresholds.sum) appendedRow.is_reliable = false;
-  if (codingThresholds.comparison_sum) appendedRow.comparison_is_reliable = false;
-
-  // !!!!!! These calculations were changed from determining statistical SIGNIFICANCE to determining statistical RELIABILITY in PR #39, variable names were NOT updated to reflect this calculation change
-  // !!!!!! These names will be updated in a broader refactor of PFF, Visit issue #57 in labs-factfinder waffle to get more info
-  appendedRow.significant = ((((difference_m) / 1.645) / Math.abs(difference_sum)) * 100) < 20;
-  appendedRow.change_significant = ((((change_m) / 1.645) / Math.abs(change_sum)) * 100) < 20;
-  appendedRow.change_percent_significant = ((((change_percent_m) / 1.645) / Math.abs(change_percent)) * 100) < 20;
-
-  return appendedRow;
-}));
-
-const invalidCompare = (compare) => {
+/*
+ * Queries postgres for current, previous, and compare data for a given
+ * profile type, set of geoids and compare geoid. Joins the data, and adds
+ * 'change' and 'difference' calculation values.
+ * @param {string} profile - The profile type
+ * @param {Array} geoids - The list of geoids for the given selected geography
+ * @param {string} compare - Integer string representing the geoid of the comparison geography
+ * @returns {Object}
+ */
+async function getProfileData(profile, geoids, compare, db) {
+  const isAggregate = geoids.length > 1;
+
+  const queryBuilder = getQueryBuilder(profile);
+
+  // get data from postgres
+  const [profileData, previousProfileData, compareProfileData] = await Promise.all([
+    db.query(queryBuilder(profile, geoids)),
+    db.query(queryBuilder(profile, geoids, /* is previous */ true)),
+    db.query(queryBuilder(profile, compare)),
+  ]);
+
+  // create Dataframe from profile data
+  let profileDF = new DataIngestor(profileData, profile, isAggregate).processRaw();
+
+  // join with previous profile data Dataframe
+  profileDF = profileDF.join(
+    new DataIngestor(previousProfileData, profile, isAggregate, /* is previous */ true).processRaw('previous'),
+    'variable',
+  );
+
+  // join with compare profile data Dataframe
+  profileDF = profileDF.join(
+    // 'compare' profiles are always NOT aggregate; they are comprised of a single geoid
+    new DataIngestor(compareProfileData, profile).processRaw('comparison'),
+    'variable',
+  );
+
+  // convert combined Dataframe to array of objects
+  // (change & difference calculations results in ad-hoc property addition,
+  // which is easier with objects than Dataframe rows)
+  const profileObj = profileDF.toCollection();
+
+  /* eslint-disable */
+  profileObj.map((row) => {
+    doChangeCalculations(row);
+    doDifferenceCalculations(row);
```

Comment:
Are these difference and change _reliability_ calculations? My understanding is that `DataIngestor` handles comparison and change calcs. So these the calcs that must be _re_calculated because of the special calculations?

Reply:
No sorry, i should clarify. The `DataIngestor` handles all `sum` and `m` recalculations for aggregate datasets, acting on each dataset (current, `previous`, `comparison`) separately. Then, all change & difference values are calculated once the three datasets are joined. This includes everything from `change_sum`, `difference_sum` etc down to `percent_significant` and `percentage_point_significant` -- these utils do it all. I can add some comments to make that more clear. 

I'm also now realizing that I don't recalculate cv in my branch!!! need to add that !